openwisp_notifications/handlers.py

```diff
@@ -175,41 +178,21 @@ def send_email_notification(sender, instance, created, **kwargs):
     # Abort if a new notification is not created
     if not created:
         return
-    # Get email preference of user for this type of notification.
-    target_org = getattr(getattr(instance, "target", None), "organization_id", None)
-
-    if instance.type and target_org:
-        try:
-            notification_setting = instance.recipient.notificationsetting_set.get(
-                organization=target_org, type=instance.type
-            )
-        except NotificationSetting.DoesNotExist:
-            return
-        email_preference = notification_setting.email_notification
-    else:
-        # We can not check email preference if notification type is absent,
-        # or if target_org is not present
-        # therefore send email anyway.
-        email_preference = True
 
     email_verified = instance.recipient.emailaddress_set.filter(
         verified=True, email=instance.recipient.email
     ).exists()
-
-    if not (email_preference and instance.recipient.email and email_verified):
+    if not email_verified or not get_user_email_preference(instance):
         return
 
-    recipient_id = str(instance.recipient.id)
-    cache_key = f"email_batch_{recipient_id}"
-
+    cache_key = f"email_batch_{instance.recipient.id}"
     cache_data = cache.get(
         cache_key,
         {
             "last_email_sent_time": None,
             "batch_scheduled": False,
             "pks": [],
             "start_time": None,
-            "email_id": instance.recipient.email,
```

Comment:
Why is this being removed?

Reply:
We already do a lookup for `User` (instance.recipient) in the celery task. Thus, we don't need to store the info in the cache.